download/download.go

```diff
@@ -247,25 +244,25 @@ func (d *Downloader) oneShot(ctx context.Context) (bool, error) {
 		if d.f != nil {
 			d.f(ctx, Update{ETag: "", Bundle: nil, Error: err, Metrics: m, Raw: nil})
 		}
-
-		return false, err
+		return err
 	}
 
 	d.etag = resp.etag
+	if d.config.Polling.LongPollingTimeoutSeconds != nil {
+		d.longPollingEnabled = resp.longPoll
+	}
```

Comment:
This can simply be 
```go
d.longPollingEnabled = resp.longPoll
```

Reply:
Before i remove the if, i prefere to be sure of something. 

Without this `if d.config.Polling.LongPollingTimeoutSeconds != nil` d.longPollingEnabled will be match if server support.  This if says: if user configure longPolling in opa. In case of user don't want/configure longpolling, there is no reason to change d.longPollingEnabled (no matter of server support or not). In other word, the server support long polling should be considered only `d.config.Polling.LongPollingTimeoutSeconds` is defined.

The if can be remove only if this two values (server support long polling, and opa is long polling configured) should be the same.  I don't think mix this two notions is good. But if you are sure no problem, let me now, i will remove the if.

Reply:
I think you're using the value of `d.longPollingEnabled ` in other parts of the code. If user switches from `regular` to `long` to `regular` polling again you'll need to update `d.longPollingEnabled ` accordingly.

Reply:
Ok i have change the code. It's seem's give good result.

Reply:
Only if the server responds with `Content-Type: application/vnd.openpolicyagent.bundles`, OPA will use long polling.

Reply:
oki thanks.